internal/tools/uvmboot/mounts.go

```diff
@@ -0,0 +1,105 @@
+//go:build windows
+
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/Microsoft/hcsshim/internal/uvm"
+	"github.com/sirupsen/logrus"
+	"github.com/urfave/cli"
+)
+
+func mountSCSI(ctx context.Context, c *cli.Context, vm *uvm.UtilityVM) error {
+	for _, m := range parseMounts(c, scsiMountsArgName) {
+		if _, err := vm.AddSCSI(
+			ctx,
+			m.host,
+			m.guest,
+			!m.writable,
+			false, // encrypted
+			[]string{},
+			uvm.VMAccessTypeIndividual,
+		); err != nil {
+			return fmt.Errorf("could not mount disk %s: %w", m.host, err)
+		} else {
+			logrus.WithFields(logrus.Fields{
+				"host":     m.host,
+				"guest":    m.guest,
+				"writable": m.writable,
+			}).Debug("Mounted SCSI disk")
+		}
+	}
+
+	return nil
+}
+
+func shareFiles(ctx context.Context, c *cli.Context, vm *uvm.UtilityVM) error {
+	switch os := vm.OS(); os {
+	case "linux":
+		return shareFilesLCOW(ctx, c, vm)
+	default:
+		return fmt.Errorf("file shares are not supported for %s UVMs", os)
+	}
+}
+
+func shareFilesLCOW(ctx context.Context, c *cli.Context, vm *uvm.UtilityVM) error {
+	for _, s := range parseMounts(c, shareFilesArgName) {
+		if err := vm.Share(ctx, s.host, s.guest, !s.writable); err != nil {
+			return fmt.Errorf("could not share file or directory %s: %w", s.host, err)
+		} else {
+			logrus.WithFields(logrus.Fields{
+				"host":     s.host,
+				"guest":    s.guest,
+				"writable": s.writable,
+			}).Debug("Shared path")
+		}
+	}
+
+	return nil
+}
+
+type mount struct {
+	host     string
+	guest    string
+	writable bool
+}
+
+// parseMounts parses the mounts stored under the cli StringSlice argument, `n`
+func parseMounts(c *cli.Context, n string) []mount {
+	if c.IsSet(n) {
+		ss := c.StringSlice(n)
```

Comment:
how does the call look like, haven't used this before?

Reply:
if the flag is `"mount"`, then it would be `uvmboot -mount a -mount b`, and `c.StringSlice("mount")` would return `["a", "b"]`